### ISSUE TITLE
Non-consecutive id error when removing tiles from collection

### ISSUE THREAD

**Issue by ablondin** [bjorn/tiled#1233]
Hi!

I'm currently teaching a class where my students have to use Tiled to develop a simple game (see http://thales.math.uqam.ca/~blondin/fr/inf3135-tp2 if you're interested, but it's in French). I provided them two maps saved with the TMX format. Below is one of those two maps (sorry, the file is a bit long):

~~~xml
<?xml version="1.0" encoding="UTF-8"?>
<map version="1.0" orientation="isometric" renderorder="right-down" width="30" height="30" tilewidth="64" tileheight="32" nextobjectid="1">
 <tileset firstgid="1" name="Tileset" tilewidth="128" tileheight="128" tilecount="15" columns="0">
  <tile id="0">
   <image width="64" height="128" source="sand.png"/>
  </tile>
  <tile id="1">
   <image width="64" height="128" source="water.png"/>
  </tile>
  <tile id="2">
   <image width="64" height="128" source="water-front.png"/>
  </tile>
  <tile id="3">
   <image width="64" height="128" source="water-left.png"/>
  </tile>
  <tile id="4">
   <image width="64" height="128" source="grass.png"/>
  </tile>
  <tile id="5">
   <image width="64" height="128" source="water-right.png"/>
  </tile>
  <tile id="6">
   <image width="64" height="128" source="earth.png"/>
  </tile>
  <tile id="8">
   <image width="64" height="128" source="stairs.png"/>
  </tile>
  <tile id="9">
   <image width="64" height="128" source="stairs-nw.png"/>
  </tile>
  <tile id="10">
   <image width="64" height="128" source="stairs-se.png"/>
  </tile>
  <tile id="11">
   <image width="64" height="128" source="stairs-sw.png"/>
  </tile>
  <tile id="12">
   <image width="64" height="128" source="tree.png"/>
  </tile>
  <tile id="13">
   <image width="64" height="128" source="rock.png"/>
  </tile>
  <tile id="16">
   <image width="128" height="128" source="house.png"/>
  </tile>
  <tile id="18">
   <image width="64" height="64" source="character.png"/>
  </tile>
 </tileset>
 <layer name="Level0" width="30" height="30">
  <data encoding="csv">
2,2,2,2,2,2,2,2,2,2,2,2,2,2,6,0,2,2,2,2,2,2,2,2,2,2,2,2,2,6,
2,2,2,2,5,5,5,5,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,5,5,5,5,5,5,6,
2,2,5,5,5,5,5,5,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,5,5,5,5,5,5,6,
2,5,5,5,5,5,5,5,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,5,5,5,5,5,5,6,
2,5,5,5,5,5,5,5,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,5,5,5,5,5,5,6,
2,5,5,5,5,5,5,1,1,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,6,
2,5,5,5,5,5,5,1,1,1,5,5,5,5,5,0,5,5,5,5,5,5,5,5,5,5,5,5,5,6,
2,5,5,5,5,5,5,1,1,1,5,5,5,5,5,0,5,5,5,5,5,5,5,5,5,2,2,5,5,6,
2,5,5,5,5,5,5,5,1,1,5,5,5,5,5,0,5,5,5,5,5,5,5,2,2,2,2,2,2,6,
2,5,5,5,5,5,5,5,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,2,1,1,1,1,1,6,
2,5,5,5,5,5,5,5,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,2,1,1,1,1,1,6,
2,5,5,5,5,5,5,5,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,2,1,1,1,1,1,6,
2,5,5,5,5,5,5,5,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,2,1,1,1,1,1,6,
3,5,5,5,5,5,5,5,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,4,1,1,1,1,1,3,
0,0,0,5,0,0,0,0,0,0,0,0,5,0,0,0,0,5,0,0,0,0,0,0,0,0,1,0,0,0,
7,5,5,5,5,5,5,5,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,2,1,1,1,1,1,6,
7,5,5,5,5,5,5,5,5,5,5,2,2,5,5,0,5,5,5,5,5,5,5,2,1,1,1,1,1,6,
7,7,7,7,5,5,5,5,5,5,5,2,2,2,5,0,5,5,5,5,5,5,5,2,2,1,1,1,1,6,
7,7,7,7,7,7,5,5,5,5,5,5,2,5,5,0,5,5,5,5,5,5,5,5,2,2,1,2,2,6,
7,7,7,7,7,7,7,5,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,5,5,2,2,2,5,6,
7,7,7,7,7,7,7,7,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,5,5,5,5,5,5,6,
7,7,7,7,7,7,7,7,5,5,5,5,5,5,5,0,5,5,5,5,5,5,5,5,5,5,5,5,5,6,
7,7,7,7,7,7,7,7,7,5,5,5,5,5,5,0,5,5,5,5,5,5,5,5,5,5,5,5,5,6,
7,7,7,7,7,7,7,7,7,5,5,5,5,5,5,0,5,5,5,5,5,5,5,5,5,5,5,5,5,6,
7,7,7,7,7,7,7,7,7,7,5,5,5,5,5,0,5,5,5,5,5,5,5,5,2,2,5,5,5,6,
7,7,7,7,7,7,7,7,7,7,7,5,5,5,5,0,5,5,5,5,5,5,5,2,2,2,2,2,5,6,
7,7,7,7,7,7,7,7,7,7,7,5,5,5,5,0,5,5,5,5,5,5,5,2,2,2,2,2,2,6,
7,7,7,7,7,7,7,7,7,7,7,7,5,5,5,5,5,5,5,5,5,5,2,2,2,2,2,2,2,6,
7,7,7,7,7,7,7,7,7,7,7,7,5,5,5,0,5,5,5,5,5,2,2,2,2,2,2,2,2,6,
7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,0,4,4,4,4,4,4,4,4,4,4,4,4,4,3
</data>
 </layer>
 <layer name="Level1" width="30" height="30" offsetx="0" offsety="-39">
  <data encoding="csv">
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,13,0,0,0,0,0,0,0,
0,0,0,0,0,13,0,0,0,0,0,17,0,0,0,0,0,0,0,0,0,0,13,0,0,0,14,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,19,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,14,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,13,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,13,0,0,0,0,0,0,0,0,0,0,0,0,0,0,14,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,13,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,12,0,0,0,
7,0,0,0,0,0,0,0,0,0,0,13,0,13,0,0,0,0,0,0,0,0,0,0,0,0,7,0,0,0,
7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,7,0,0,0,
7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,7,0,0,0,
7,7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,9,0,0,0,
7,7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,7,7,10,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,7,7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,7,7,7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,13,0,0,0,0,0,0,0,
7,7,7,7,7,7,7,0,0,7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,7,7,7,7,7,7,7,7,7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,7,7,7,7,7,7,7,7,7,7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
</data>
 </layer>
 <layer name="Level2" width="30" height="30" offsetx="0" offsety="-78">
  <data encoding="csv">
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
12,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,7,7,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
7,7,7,7,7,7,7,10,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
</data>
 </layer>
 <layer name="Level3" width="30" height="30" offsetx="0" offsety="-117">
  <data encoding="csv">
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
14,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
14,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
</data>
 </layer>
</map>
~~~

As you might notice, the tiles `id` values are not consecutive (`7`, `14`, `15` and `17`) are missing and some of my students told me that they couldn't properly load the file, getting an error about **non-consecutive tile id**. Strangely enough, I don't get the error on my own machine (which is a MacBook Pro), so that it's hard for me to give more details about it.

I was wondering if it was possible to change the id of the tiles so that they become consecutive. I'm thinking about writing a Python script to do it, but maybe Tiled has already provided a better solution for this problem? I've searched around on the issues list and on Google, without success.

Note that the reason why I have nonconsecutive tile ids is due to the fact that I removed tiles while I was editing the map (some images were produced with another tool and I had to modify them). Now, I understand I should have avoided adding/removing tiles (which made Tiled crash a few times), but it seemed to be ok until I exported the file and other people tried to read it.

This seem to be somewhat related to issue #1118, but I don't see how I can correct the problem now, if it's possible.

Any help is welcome!

**Comment by ablondin** [comment]
In the same spirit, why not start with `<tile id="1">` systematically instead of `<tile id="0">` since the *empty tile* is always 0? It would make the mapping more obvious in the file. Now, we have to substract 1 to each value between the `<data>` and `</data>` to get the matching tile id, and when there are nonconsecutive, the correspondance is even more complicated.

**Comment by Seanba** [comment]
This seems to be the problem to my eyes: <tileset firstgid="1" name="Tileset
" tilewidth="128" tileheight="128" tilecount="15" columns="0">

There are only 15 tiles in the tileset yet the layers in the map are
referencing a tile with and ID of 17. To me, that sounds like some tiles
were removed from the tileset but are still being used by the layer?

**Comment by ablondin** [comment]
@Seanba: I agree, there are 15 tiles, but there are gaps in the ids. This is because I removed tiles when editing the map. But each used tile in the layers are valid. For instance, the id `14` does not exist, but it is not used neither in the layers, since `14+1=15` does not appear anywhere.

My point is: Tiled shouldn't let us build corrupted maps. I didn't edit the file, I just saved it. This seems to be a bug as far as I can see. Of course, I shouldn't have added/removed tiles so often (this is my mistake), but it would be cleaner if Tiled forbids us from doing it, or provide a way to *uncorrupt* the file.

**Comment by bjorn** [comment]
> As you might notice, the tiles id values are not consecutive (7, 14, 15 and 17) are missing and some of my students told me that they couldn't properly load the file, getting an error about `non-consecutive tile id`. Strangely enough, I don't get the error on my own machine

You correctly referenced issue #1118 as related to this issue. To fix that issue, I had to remove the requirement of tile IDs to be consecutive. It's unfortunate that older versions of Tiled require this. The simple solution to your problem should be to make sure all your students are using Tiled 0.15.

So, there's nothing corrupt about your map. It's just that maps with image collection tilesets from which images have been deleted are not backwards compatible. The reason I couldn't just collapse those gaps anymore is because that could invalidate tile references in many maps, including maps not currently open in Tiled (in case of modifying an external tileset).

> In the same spirit, why not start with <tile id="1"> systematically instead of <tile id="0"> since the empty tile is always 0?

Here you're confusing global tile IDs with the local tile IDs. The local ID starts at 0 because C arrays start at 0. For global tile IDs, 0 always means "no tile", and 1 refers to the first tile in the tileset. The mapping from global ID to local ID is not done by simply subtracting 1, but by subtracting the `firstgid` value of the tileset. This is the global ID of the first tile in the tileset. This allows a map to refer to tiles from multiple tilesets.

> Now, I understand I should have avoided adding/removing tiles (which made Tiled crash a few times)

I should of course fix that crash, rather than you needing to avoid using basic Tiled features. Did you have multiple maps open? Can you reproduce the crash?

> I'm currently teaching a class where my students have to use Tiled to develop a simple game (see http://thales.math.uqam.ca/~blondin/fr/inf3135-tp2 if you're interested, but it's in French).

That looks like a very complete course on tile maps. Nice!

**Comment by ablondin** [comment]
Hi, Bjorn,

Thank you for your quick and thorough reply. You answered all my questions and now I understand better how `id` and `firstgid` works. I'm glad the file is not corrupted and I'll let my students know about using more recent versions of Tiled. As far as I'm concerned, this issue can be closed.

To answer your last question, I just tried reproducing the crash, without success. However, I found another problem: it seems that if you remove the active tile from the tileset, you can still use it in the editor (and everything works without crashing). I guess it would be safer to deactivate the tile if it is removed. Note that I'm using Version v0.15.1-62-g2406cdb, so maybe the bug has been corrected since then? Anyway, if I ever succeed in reproducing the bug, I'll let you know.

Thanks again.

**Comment by bjorn** [comment]
> However, I found another problem: it seems that if you remove the active tile from the tileset, you can still use it in the editor (and everything works without crashing).

I'm not sure currently if that still happens, but I should make sure this is fixed before releasing Tiled 0.16. Thanks for pointing it out!

**Comment by bjorn** [comment]
> I'm not sure currently if that still happens, but I should make sure this is fixed before releasing Tiled 0.16. Thanks for pointing it out!

I didn't end up fixing this for Tiled 0.16 because I wasn't sure how to prevent it without potentially breaking other things, and I didn't want to delay the release further. I've opened issue #1243 about this, so that I can close this one.
